### PR TITLE
habtm for product promotion rules

### PR DESCRIPTION
**Pull request by BenMorganIO**

### CODE REVIEW

**Comment by jhawthorn** [comment]
:+1:

File: core/app/models/spree/product_promotion_rule.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+module Spree
+  class ProductPromotionRule < Spree::Base
+    belongs_to :product
```
Comment (jhawthorn) [inline core/app/models/spree/product_promotion_rule.rb]: Should we have an inverse of here, or is that detected successfully?

Comment (BenMorganIO) [inline core/app/models/spree/product_promotion_rule.rb]: I think it can get detected.

**Comment by jhawthorn** [comment]
I checked and it is. LGTM

**Comment by magnusvk** [comment]
:+1:

**Comment by BenMorganIO** [comment]
Left a similar comment in #317.

No dependent destroy was added. Backtrack and add referential integrity as well?

**Pull request merged**